sympy/tensor/tests/test_indexed.py

```diff
@@ -203,6 +203,26 @@ def test_IndexedBase_shape():
     raises(IndexException, lambda: F[i, j])
 
 
+def test_IndexedBase_assumptions():
+    i = Symbol('i', integer=True)
+    a = Symbol('a')
+    A = IndexedBase(a, positive=True)
+    b = Indexed(a, i, positive=True)
+    for c in (A, b, A[i]):
+        assert c.is_real
+        assert c.is_complex
+        assert not c.is_imaginary
+        assert c.is_nonnegative
+        assert c.is_nonzero
+        assert c.is_commutative
+        assert log(exp(c)) == c
+
+    assert A != IndexedBase(a)
+    assert A == IndexedBase(a, positive=True, real=True)
```

Comment:
is `real` redundant here?

Reply:
Yes, kind of, but on purpose. I left it in to verify that the assumptions were transformed into the same set, even if a user specifies redundant assumptions. Maybe unnecessary/unclear. Would you like me to change it? Perhaps as a separate test instead?

Reply:
it's ok